issue.go

```diff
@@ -123,53 +134,92 @@ func DeleteIssue(db *sql.DB, id int) error {
 	return nil
 }
 
-func getIssueHandler(c *gin.Context, db *sql.DB) {
+func postIssueHandler(c *gin.Context, db *sql.DB) {
 
-	id := c.Param("id")
+	userId, err := authorization(c, db)
+	if err != nil {
+		return
+	}
+	username := c.Param("owner")
+	repoName := c.Param("repo")
+	var repoId string
+	row := db.QueryRow(`SELECT repos.id FROM repos JOIN users ON repos.user_id= users.id 
+						   WHERE repos.name=$1 AND users.username=$2`, repoName, username)
+	err = row.Scan(&repoId)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
 
-	Id, err := strconv.Atoi(id)
+	issue := Issue{}
+	err = c.BindJSON(&issue)
 	if err != nil {
+		fmt.Println(err)
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
 
-	issue, err := GetIssue(db, Id)
+	issue.UserId = userId
+	issue.RepoId = repoId
 
+	_, _, err = CreateIssue(db, issue)
 	if err != nil {
+		fmt.Println(err)
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
-	c.JSON(http.StatusOK, issue)
 
-	// c.Status(http.StatusNoContent)
+	c.Status(http.StatusCreated)
 
 }
 
-func postIssueHandler(c *gin.Context, db *sql.DB) {
+func putIssueHandler(c *gin.Context, db *sql.DB) {
 
-	issue := Issue{}
-	err := c.BindJSON(&issue)
+	_, err := authorization(c, db)
+	if err != nil {
+		return
+	}
+	username := c.Param("owner")
+	repoName := c.Param("repo")
+	issueNumber := c.Param("issue_number")
+
+	var Id int
+	row := db.QueryRow(`WITH repo_cte AS (select repos.id FROM repos JOIN users ON repos.user_id= users.id 
+                		WHERE repos.name= $1 AND users.username= $2) select id from issues where repo_id 
+                		in (select id from repo_cte) and issue_number=$3`, repoName, username, issueNumber)
+	err = row.Scan(&Id)
 	if err != nil {
 		fmt.Println(err)
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
 
-	_, _, err = CreateIssue(db, issue)
+	issue := Issue{}
+	err = c.BindJSON(&issue)
 	if err != nil {
 		fmt.Println(err)
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
 
-	c.Status(http.StatusCreated)
-
-}
+	query := psql.Update("issues")
+	if issue.Status != "" {
+		query = query.Set("status", issue.Status)
+	}
+	if issue.Title != "" {
+		query = query.Set("title", issue.Title)
+	}
+	if issue.Body != "" {
+		query = query.Set("body", issue.Body)
+	}
+	if issue.Pinned != "" {
+		query = query.Set("pinned", issue.Pinned)
+	}
```

Comment:
pinned should not be done here in PUT.

Have POST /pin  and DELETE /pin. 

Check Condition POST: When you pin issue, you have to check there are no more than 2 pinned issues already.

Reply:
done